src/components/pages/profile/[name]/Profile.tsx

```diff
@@ -150,6 +145,31 @@ const ProfileContent = ({ nameDetails, isSelf, isLoading, name }: Props) => {
 
   const selfAbilities = useSelfAbilities(address, name)
 
+  useEffect(() => {
+    if (
+      name !== profile?.decryptedName &&
+      profile?.decryptedName &&
+      !isSelf &&
+      getEncryptedLabelAmount(normalisedName) > getEncryptedLabelAmount(profile.decryptedName)
+    ) {
+      router.replace(`/profile/${profile.decryptedName}`, { shallow: true, maintainHistory: true })
+    } else if (
```

Comment:
Can you move this into its own named hook or add a brief comment on what it's doing